surfactant/cmd/tui.py

```diff
@@ -0,0 +1,484 @@
+    def add_config_entry(self):
+        self.config_entries.append(ConfigEntry(self.config_number))
+        self.mount(self.config_entries[-1], before="#add_config_entry")
+        self.config_number += 1
+
+    @textual.on(textual.widgets.Button.Pressed, "#save")
+    def save(self):
+        to_save = []
+        for entry in self.config_entries:
+            to_save.append({})
+            write_to = to_save[-1]
+            archive = entry.archive.input_path
+            if len(archive) > 0:
+                write_to["archive"] = archive
+            install_prefix = entry.install_prefix.value
+            if len(install_prefix) > 0:
+                write_to["install_prefix"] = install_prefix
+            write_to["extractPaths"] = []
+            for path in entry.extract_paths.input_paths:
+                if path.active:
+                    write_to["extractPaths"].append(path.path_selector.input_path)
+        file_to_save = self.config_input.input_path + "/" + self.config_name.value
+        try:
+            with open(file_to_save, "w") as f:
+                f.write(json.dumps(to_save, indent=2))
+        except IsADirectoryError:
+            self.app.notify(f"Could not write to {file_to_save}")
+            return
+        self.app.notify(f"Wrote config to {file_to_save}")
+
+    @textual.on(textual.widgets.Button.Pressed, "#load")
+    def load(self):
+        file_to_load = self.config_input.input_path + "/" + self.config_name.value
+        try:
+            with open(file_to_load, "r") as config_file:
+                js = json.load(config_file)
+        except (FileNotFoundError, IsADirectoryError):
+            self.app.notify(f"Could not find file {file_to_load}")
+            return
+        except json.JSONDecodeError:
+            self.app.notify(f"Error when parsing JSON in {file_to_load}")
+            return
+        # Delete all other entries
+        for ce in self.config_entries:
+            ce.remove()
+        self.config_entries = []
+        self.config_number = 1
+        for entry in js:
+            self.config_entries.append(ConfigEntry(self.config_number))
+            self.config_number += 1
+            cur_entry = self.config_entries[-1]
+            if "archive" in entry:
+                cur_entry.archive.input_path = entry["archive"]
+            if "extractPaths" in entry:
+                for ep in entry["extractPaths"]:
+                    cur_entry.extract_paths.add_path(ep)
+            if "installPrefix" in entry:
+                cur_entry.install_prefix.value = entry["installPrefix"]
+        for entry in self.config_entries:
+            self.mount(entry, before="#add_config_entry")
+
+
+# TODO: Rewrite to use ContentSwitcher?
+class TUI(textual.app.App):
+    """An app for running Surfactant commands"""
+
+    BINDINGS = [
+        ("d", "toggle_dark", "Toggle dark mode"),
+        ("q", "quit", "Quit"),
+    ]
+    TITLE = "Surfactant TUI"
+    CSS_PATH = "../web-files/tui.tcss"
```

Comment:
Can we move this to a different folder that isn't web-specific? Maybe along the lines of "ui-resources" or "ui-data"?